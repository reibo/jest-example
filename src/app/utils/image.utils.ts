import {Image} from '../snapshot/image.model';

const IMAGE_SOURCE = 'https://source.unsplash.com/random/400x300';

function generateRandomUrl(id: number) {
  return `url("${IMAGE_SOURCE}?sig=${id})`;
}

function getRows(id: number, totalElements: number, maxCols: number) {
  return (totalElements - id) % maxCols || 1;
}


function getCols(id: number, maxCols: number) {
  return Math.floor((id % maxCols) / 3 + 1);
}

export function listImages(totalImages: number, maxCols: number): Image[] {
  const images: Image[] = [];
  for (let i = 0; i < totalImages; i++) {
    images.push({
      id: i,
      src: generateRandomUrl(i),
      rows: getRows(i, totalImages, maxCols),
      cols: getCols(i, maxCols)
    });
  }
  return images;
}
