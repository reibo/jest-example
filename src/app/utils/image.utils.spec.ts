import {listImages} from './image.utils';

describe('image.utils', () => {
  describe('on list images', () => {
    it('should return a complete list', () => {
      expect(listImages(3, 2)).toMatchSnapshot();
    });
  });

  // @ts-ignore
  describe.each([[1, 1], [5, 2], [3, 2]])(
    '.listImages(%i, %i)',
    (totalElement: number, maxCols: number) => {
      it(`match snapshot`, () => {
        expect(listImages(totalElement, maxCols)).toMatchSnapshot();
      });
    }
  );

  describe.each([[1, 1, 2], [1, 2, 3], [2, 1, 3]])(
    '.add(%i, %i)',
    (a, b, expected) => {
      test(`returns ${expected}`, () => {
        expect(a + b).toBe(expected);
      });

      test(`returned value not be greater than ${expected}`, () => {
        expect(a + b).not.toBeGreaterThan(expected);
      });

      test(`returned value not be less than ${expected}`, () => {
        expect(a + b).not.toBeLessThan(expected);
      });
    }
  );


  test.todo('add test getRows');
});
