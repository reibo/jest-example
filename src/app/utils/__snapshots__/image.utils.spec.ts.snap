// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`image.utils .listImages(1, 1) match snapshot 1`] = `
Array [
  Object {
    "cols": 1,
    "id": 0,
    "rows": 1,
    "src": "url(\\"https://source.unsplash.com/random/400x300?sig=0)",
  },
]
`;

exports[`image.utils .listImages(3, 2) match snapshot 1`] = `
Array [
  Object {
    "cols": 1,
    "id": 0,
    "rows": 1,
    "src": "url(\\"https://source.unsplash.com/random/400x300?sig=0)",
  },
  Object {
    "cols": 1,
    "id": 1,
    "rows": 1,
    "src": "url(\\"https://source.unsplash.com/random/400x300?sig=1)",
  },
  Object {
    "cols": 1,
    "id": 2,
    "rows": 1,
    "src": "url(\\"https://source.unsplash.com/random/400x300?sig=2)",
  },
]
`;

exports[`image.utils .listImages(5, 2) match snapshot 1`] = `
Array [
  Object {
    "cols": 1,
    "id": 0,
    "rows": 1,
    "src": "url(\\"https://source.unsplash.com/random/400x300?sig=0)",
  },
  Object {
    "cols": 1,
    "id": 1,
    "rows": 1,
    "src": "url(\\"https://source.unsplash.com/random/400x300?sig=1)",
  },
  Object {
    "cols": 1,
    "id": 2,
    "rows": 1,
    "src": "url(\\"https://source.unsplash.com/random/400x300?sig=2)",
  },
  Object {
    "cols": 1,
    "id": 3,
    "rows": 1,
    "src": "url(\\"https://source.unsplash.com/random/400x300?sig=3)",
  },
  Object {
    "cols": 1,
    "id": 4,
    "rows": 1,
    "src": "url(\\"https://source.unsplash.com/random/400x300?sig=4)",
  },
]
`;

exports[`image.utils on list images should return a complete list 1`] = `
Array [
  Object {
    "cols": 1,
    "id": 0,
    "rows": 1,
    "src": "url(\\"https://source.unsplash.com/random/400x300?sig=0)",
  },
  Object {
    "cols": 1,
    "id": 1,
    "rows": 1,
    "src": "url(\\"https://source.unsplash.com/random/400x300?sig=1)",
  },
  Object {
    "cols": 1,
    "id": 2,
    "rows": 1,
    "src": "url(\\"https://source.unsplash.com/random/400x300?sig=2)",
  },
]
`;
